packages/instantsearch.js/src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.ts

```diff
@@ -316,6 +316,48 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/infinite-hi
     ).toHaveBeenCalledWith(expect.objectContaining({ page: 3 }));
   });
 
+  it('goes to previous page correctly when state is mismatched', () => {
+    const renderFn = jest.fn();
+    const makeWidget = connectInfiniteHits(renderFn);
+    const widget = makeWidget({});
+
+    const helper = algoliasearchHelper(createSearchClient(), '', {
+      page: 4,
+    });
+    helper.overrideStateWithoutTriggeringChangeEvent = jest.fn(() => helper);
+    helper.searchWithoutTriggeringOnStateChange = jest.fn();
+
+    const mainHelper = algoliasearchHelper(createSearchClient(), '', {
+      page: 4,
+      query: 'test', // arbitrary change to state to force cache mismatch
+    });
+    mainHelper.overrideStateWithoutTriggeringChangeEvent = jest.fn(
+      () => mainHelper
+    );
+    mainHelper.searchWithoutTriggeringOnStateChange = jest.fn();
```

Comment:
and here you could reset, as it seems to be set twice

Reply:
Actually don't need to mock anything for the main helper since we only care about it's state.